packages/aws-cdk-lib/aws-route53/lib/hosted-zone-ref.ts

```diff
@@ -36,8 +37,11 @@ export interface IHostedZone extends IResource {
 
   /**
    * Grant permissions to add delegation records to this zone
+   *
+   * @param grantee grantee to receive the permissions
+   * @param names specify to restrict the delegation to a specific set of names
    */
-  grantDelegation(grantee: iam.IGrantable): iam.Grant;
+  grantDelegation(grantee: iam.IGrantable, names?: DelegationGrantNames): iam.Grant;
```

Comment:
I think we should use a property bag. We only have once chance to add an optional prop to a public API with backwards compatibility, so lets keep the door open for future additions too. It has the additional benefit of forcing users to supply the prop name with their input, which makes things clearer.

`grantDelegation(grantee: iam.IGrantable, { nameLike?: string, nameEquals?: string }): iam.Grant;`

Reply:
Acknowledged. Does this look good to you?

```ts
interface GrantDelegationProps {
  readonly name?: {
    readonly equals: string[];
  } | {
    readonly like: string[];
  };
}
```

Reply:
ideally i want

```ts
interface GrantDelegationProps {
  readonly nameEquals?: string[];
  readonly nameLike?: string[];
}
```

And then somewhere else make sure that `nameEquals` and `nameLike` are not both set at once.